src/main/resources/static/js/merge.js

```diff
@@ -2,28 +2,58 @@ let currentSort = {
   field: null,
   descending: false,
 };
+//New Array to keep track of unique id
+let filesWithUniqueId = [];
+let processedFiles = [];
 
 document.getElementById("fileInput-input").addEventListener("change", function () {
-  var files = this.files;
+  var files = Array.from(this.files).map(file => {
+    return {
+      file: file,
+      uniqueId: file.name + Date.now()
+    };
+  });
+  filesWithUniqueId = files;
   displayFiles(files);
 });
+//Get Files Updated Event from FileInput
+document.addEventListener("filesUpdated", function (e) {
+  filesWithUniqueId = e.detail;
+  displayFiles(filesWithUniqueId);
+});
+
 
-/**
- * @param {FileList} files
- */
 function displayFiles(files) {
   const list = document.getElementById("selectedFiles");
 
   while (list.firstChild) {
     list.removeChild(list.firstChild);
   }
 
+  // Clear the processedFiles array
+  processedFiles = [];
+
   for (let i = 0; i < files.length; i++) {
     const item = document.createElement("li");
     item.className = "list-group-item";
+    item.dataset.id = files[i].uniqueId; // Assign the uniqueId to the list item
+    const fileNameDiv = document.createElement("div");
+    fileNameDiv.className = "filename";
+    fileNameDiv.textContent = files[i].file.name;
+
+    // Check for duplicates and add a warning if necessary
+    const duplicateFiles = files.filter(file => file.file.name === files[i].file.name);
+    if (duplicateFiles.length > 1) {
+      const warning = document.createElement("span");
+      warning.className = "duplicate-warning";
+      warning.textContent = "(Duplicate)";
```

Comment:
Should be translatable and dynamic

Reply:
will add as todo before next release

Reply:
Somehow did not think about that. I can try and do that as well.